client/me/purchases/manage-purchase/payment-info-block.tsx

```diff
@@ -31,6 +31,16 @@ export default function PaymentInfoBlock( {
 			card.meta?.find( ( meta ) => meta.meta_key === 'is_backup' )?.meta_value
 	);
 
+	if ( ! isRechargeable( purchase ) && hasPaymentMethod( purchase ) ) {
```

Comment:
🤔  Is there a case that we might want to see one of the other text options displayed instead of this one?

For example, what if the purchase was made with a non-rechargable payment method, but is also included with a plan? Because this condition comes first, it will always display this text instead. I'm not sure if that's what we want. Perhaps a better example is: what happens if the person used credits to make the purchase? Credits are (probably?) not rechargeable either, but the condition that displays "Credits" is several blocks below this one. We might want to move it down so that those cases are covered as well. (These are the sorts of things you can double-check with unit tests.)

Reply:
> but the condition that displays "Credits" is several blocks below this one. We might want to move it down so that those cases are covered as well. 

Fair point, I have moved it just before the final default condition that displays `None`